lib/modules/versioning/deb/index.spec.ts

```diff
@@ -0,0 +1,109 @@
+    ${'1:1:1:2-1'}                                 | ${true}
+    ${'1:a:b:c:2-1'}                               | ${true}
+    ${'1:3_3.2-1'}                                 | ${false}
+    ${'1:3!3.2-1'}                                 | ${false}
+    ${'1:3/3.2-1'}                                 | ${false}
+    ${'1.0-3_2'}                                   | ${false}
+    ${'1.0-3!3'}                                   | ${false}
+    ${'1.0-3/3'}                                   | ${false}
+    ${'1.0+ä1-1'}                                  | ${false}
+    ${'1,0-1'}                                     | ${false}
+    ${'2:1.1-1'}                                   | ${true}
+    ${'1.1.1-0debian1'}                            | ${true}
+    ${'1.1.1+really1.1.2-0debian1'}                | ${true}
+    ${'2.31-13+deb11u5'}                           | ${true}
+    ${'1:4.14+20190211-1ubuntu1'}                  | ${true}
+    ${'2.7.7+dfsg-12'}                             | ${true}
+    ${'9.5.0-1ubuntu1~22.04'}                      | ${true}
+    ${'5:20.10.17~3-0~ubuntu-focal'}               | ${true}
+    ${'1:6.0.1r16-1.1build1'}                      | ${true}
+    ${'2:102.12+LibO7.3.7-0ubuntu0.22.04.1'}       | ${true}
+    ${'1:2.20.1-1~bpo9+1'}                         | ${true}
+    ${'v1.4'}                                      | ${true}
+    ${'3.5.0'}                                     | ${true}
+    ${'4.2.21.Final'}                              | ${true}
+    ${'0.6.5.1'}                                   | ${true}
+    ${'20100527'}                                  | ${true}
+    ${'2.1.0-M3'}                                  | ${true}
+    ${'4.3.20.RELEASE'}                            | ${true}
+    ${'1.1-groovy-2.4'}                            | ${true}
+    ${'0.8a'}                                      | ${true}
+    ${'3.1.0.GA'}                                  | ${true}
+    ${'3.0.0-beta.3'}                              | ${true}
+    ${'foo'}                                       | ${true}
+    ${'1.2.3.4.5.6.7'}                             | ${true}
+    ${'0a1b2c3'}                                   | ${true}
+    ${'0a1b2c3d'}                                  | ${true}
+    ${'0a1b2c3d4e5f6a7b8c9d0a1b2c3d4e5f6a7b8c9d'}  | ${true}
+    ${'0a1b2c3d4e5f6a7b8c9d0a1b2c3d4e5f6a7b8c9d0'} | ${true}
+    ${'0a1b2C3'}                                   | ${true}
+    ${'0z1b2c3'}                                   | ${true}
+    ${'0A1b2c3d4e5f6a7b8c9d0a1b2c3d4e5f6a7b8c9d'}  | ${true}
+    ${'123098140293'}                              | ${true}
+  `('isValid("$version") === $expected', ({ version, expected }) => {
+    expect(deb.isValid(version)).toBe(expected);
+  });
+
+  test.each`
+    a          | b          | expected
+    ${'2.4'}   | ${'2.4'}   | ${true}
+    ${'2.4.0'} | ${'2.4.0'} | ${true}
+    ${'2.4.0'} | ${'2.4'}   | ${false}
+    ${'2.4.1'} | ${'2.4'}   | ${false}
+    ${'2.4.2'} | ${'2.4.1'} | ${false}
```

Comment:
shouldn't you test some more complex versions too here?

Reply:
I'd specifically like to see more tests with `~`.

From the spec:

> The lexical comparison is a comparison of ASCII values modified so that all the letters sort earlier than all the non-letters and so that a tilde sorts before anything, even the end of a part. For example, the following parts are in sorted order from earliest to latest: `~~`, `~~a`, `~`, the empty part, `a`.